### PR TITLE
Initial version for Python SDK v0.1.

### PR DESCRIPTION
**Pull request by huxuan**
- Python SDK code.
- Docstrings with contents mostly from official [API reference](https://dev.projectoxford.ai/docs/services/563879b61984550e40cbbe8d) with modification to make it more Pythonic.
- Unittest mostly for basic functionality. (Add robust unittest in TODO section of `README.md`)
- Various configuration files:
  * `.gitignore` for `git`
  * `.pylintrc` for `pylint`
  * `MANIFEST.in` and `setup.py` for `pypi`
  * `setup.cfg` for `nosetests`
  * `Makefile` and `requirements.txt` for easy development.

### CODE REVIEW

**Comment by ryanga** [comment]
@cristiancanton is added to the review. #Closed

**Comment by ryanga** [comment]
@wangjun-microsoft is added to the review. #Closed

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+"""
+from setuptools import find_packages
+from setuptools import setup
+
+README = 'README.md'
+
+
+def readme():
+    """Parse README for long_description."""
+    return open(README).read()
```
Comment (ryanga) [inline setup.py]: setup didn't run successfully for me until I manually specified the encoding of the readme file. I'm using python 3.5.2 on windows

    return open(README, encoding="utf8").read()

File: cognitive_face/tests/config.sample.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- Copy `config.sample.py` to `config.py`.
+- Assign the `KEY` with a valid Subscription Key.
+"""
+
+# Subscription Key for calling the Cognitive Face API.
+KEY = ''
+
+# Time (in seconds) for sleep between each call to avoid exceeding quota.
+# Default to 3 as free subscription have limit of 20 calls per minute.
+SLEEP_TIME = 3
```
Comment (ryanga) [inline cognitive_face/tests/config.sample.py]: >SLEEP_TIME [](start = 0, length = 10)

This is used as `TIME_SLEEP` elsewhere

**Comment by ryanga** [comment]
:clock1:

File: cognitive_face/tests/config.sample.py (hunk 2 above)
Comment (huxuan) [inline cognitive_face/tests/config.sample.py]: Done. Sorry for the silly typo.

File: setup.py (hunk 1 above)
Comment (huxuan) [inline setup.py]: Hi @ryanga,  Thanks for your review, seems this problem is caused by the default locale of the windows [1] [2]. I was using `Bash on Ubuntu on Windows` so I didn't encounter this problem. As `encoding` is not a valid argument for `open()` in Python 2, I use `io.open()` instead according to the suggestions from Python porting [3]. If there is any further problem, feel free to tell me.

[1] https://wiki.python.org/moin/PrintFails#Windows
[2] http://python-notes.curiousefficiency.org/en/latest/python3/text_file_processing.html#files-in-an-ascii-compatible-encoding-best-effort-is-acceptable
[3] https://docs.python.org/3/howto/pyporting.html

File: cognitive_face/face.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        'returnFaceLandmarks': landmarks and 'true' or 'false',
+        'returnFaceAttributes': attributes,
+    }
+
+    return util.request('POST', url, headers=headers, params=params, json=json,
+                        data=data)
+
+
+def find_similars(face_id, face_list_id=None, face_ids=None, max_candidates=20,
+                  mode='matchPerson'):
```
Comment (wangjun-microsoft) [inline cognitive_face/face.py]: Enumerate the mode to be matchPerson and matchFace

File: cognitive_face/face.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        'returnFaceId': face_id and 'true' or 'false',
+        'returnFaceLandmarks': landmarks and 'true' or 'false',
+        'returnFaceAttributes': attributes,
+    }
+
+    return util.request('POST', url, headers=headers, params=params, json=json,
+                        data=data)
+
+
+def find_similars(face_id, face_list_id=None, face_ids=None, max_candidates=20,
```
Comment (wangjun-microsoft) [inline cognitive_face/face.py]: split this to be two function may be better. one use face_ids, another use face_list_id

Comment (wangjun-microsoft) [inline cognitive_face/face.py]: max_candidates_return may be better than max_andidates

File: cognitive_face/face.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        'personGroupId': person_group_id,
+        'faceIds': face_ids,
+        'maxNumOfCandidatesReturned': max_candidates,
+        'confidenceThreshold': threshold,
+    }
+
+    return util.request('POST', url, json=json)
+
+
+def verify(face_id, another_face_id=None, person_group_id=None,
```
Comment (wangjun-microsoft) [inline cognitive_face/face.py]: split this function to two: face to face verify and face to person verify.

File: cognitive_face/person_group.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+    Returns:
+        The person group's information.
+    """
+    url = 'persongroups/{}'.format(person_group_id)
+
+    return util.request('GET', url)
+
+
+def status(person_group_id):
```
Comment (wangjun-microsoft) [inline cognitive_face/person_group.py]: use getstatus instead of status

File: cognitive_face/tests/__init__.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+File: __init__.py
+Description: Unittests for Python SDK of the Cognitive Face API.
```
Comment (wangjun-microsoft) [inline cognitive_face/tests/__init__.py]: is the call quota enough for all the tests?

Comment (wangjun-microsoft) [inline cognitive_face/tests/__init__.py]: Remind the user to paste the subscription key before install? OR paste the subscription key as the install parameter?

support A key installation??

File: cognitive_face/face.py (hunk 4 above)
Comment (huxuan) [inline cognitive_face/face.py]: Done.

---
In reply to: [76578562](https://github.com/Microsoft/Cognitive-Face-Python/pull/2#discussion_r76578562) [](ancestors = 76578562)

File: cognitive_face/tests/__init__.py (hunk 7 above)
Comment (huxuan) [inline cognitive_face/tests/__init__.py]: As all the required data is setup before running unittests, we avoid creating duplicate data. Also thanks to the `wait()` method in `tests.util`, free subscription can handle the unittests really well, with running for about 5 mins.

---
In reply to: [76582368](https://github.com/Microsoft/Cognitive-Face-Python/pull/2#discussion_r76582368) [](ancestors = 76582368)

File: cognitive_face/person_group.py (hunk 6 above)
Comment (huxuan) [inline cognitive_face/person_group.py]: Done by renaming it to `get_status()`.

---
In reply to: [76579965](https://github.com/Microsoft/Cognitive-Face-Python/pull/2#discussion_r76579965) [](ancestors = 76579965)

**Pull request merged**